ADAL/tests/XCTestCase+TestHelperMethods.h

```diff
@@ -40,6 +40,13 @@
 
 #define TEST_SIGNAL dispatch_semaphore_signal(_dsem)
 #define TEST_WAIT dispatch_semaphore_wait(_dsem, DISPATCH_TIME_FOREVER)
+#define TEST_WAIT_NOT_BLOCKING_MAIN_QUEUE \
```

Comment:
I'm undecided if reusing the semaphore like this is a anti-pattern or not, the code got here because of quick mutations from previous even worse state, but it might be worth looking at this during QP to see if this is a pattern we want to continue.